homeassistant/components/iskra/sensor.py

```diff
@@ -210,37 +205,30 @@ async def async_setup_entry(
                 sensors.append(ATTR_PHASE3_POWER)
                 sensors.append(ATTR_PHASE3_CURRENT)
 
-        entities = [
-            IskraSensor(coordinator, base_device, entry, description)
+        entities += [
+            IskraSensor(coordinator, description)
             for description in SENSOR_TYPES
             if description.key in sensors
         ]
 
-        async_add_entities(entities)
+    async_add_entities(entities)
 
 
-class IskraSensor(IskraDevice, CoordinatorEntity, SensorEntity):
+class IskraSensor(IskraEntity, SensorEntity):
     """Representation of a Sensor."""
 
-    _attr_has_entity_name = True
     entity_description: IskraSensorEntityDescription
-    coordinator: IskraDataUpdateCoordinator
 
     def __init__(
         self,
         coordinator: IskraDataUpdateCoordinator,
-        gateway,
-        config_entry,
         description: IskraSensorEntityDescription,
     ) -> None:
         """Initialize the sensor."""
-
-        self.coordinator = coordinator
         self.entity_description = description
         self._attr_unique_id = f"{coordinator.device.serial}_{description.key}"
         self._unique_id = self._attr_unique_id
```

Comment:
why?

Reply:
removed, misunderstood, checked HA's Entity class code and its clear to me now sorry